controllers/workspace/http.go

```diff
@@ -64,3 +76,33 @@ func setupHttpClients() {
 		Timeout:   500 * time.Millisecond,
 	}
 }
+
+func readCertificates(k8s client.Client, logger logr.Logger) (map[string]string, bool) {
+	configmapRef := config.GetGlobalConfig().Routing.TLSCertificateConfigmapRef
```

Comment:
I mentioned in another comment asking whether we want to be setting up the http client on a per-workspace basis (i.e. during every workspace reconcile). Doing so doesn't seem to have any advantage to me.

However, if we do want to setup the http client on a per-workspace basis, then here we might as well take advantage of the fact that any given workspace might use a [custom/external DevWorkspaceOperatorConfig](https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#setting-an-alternate-configuration-for-a-workspace). This means, users could provide different certs for different workspaces.

I don't think this is actually necessary or that useful however, and in the case of Che, every workspace uses the Che-owned external DevWorkspaceOperator Config. This means Che users have to decide between having their workspaces have Che settings propagated automatically to their workspaces, and using their own user-provided external DevWorkspaceOperatorConfig.

Reply:
After seeing @tolusha's [comment](https://github.com/devfile/devworkspace-operator/pull/1259#issuecomment-2098231359), I think we might as well take advantage of the external workspace DWOC here. 

Rather than getting the config from the global config `config.GetGlobalConfig().Routing.TLSCertificateConfigmapRef`, we can move [`setupHttpClients()`](https://github.com/devfile/devworkspace-operator/pull/1259/files#diff-05e72230161a4d41c5a27fb453041c78839c193a9ddcb5066b1cb62c5a399529R117) so that it is called immediately after the [external DWOC is resolved](https://github.com/devfile/devworkspace-operator/blob/main/controllers/workspace/devworkspace_controller.go#L132-L143). Then we can pass in the config to `setupHttpClients()`.

Reply:
moved [setupHttpClients()](https://github.com/devfile/devworkspace-operator/pull/1259/files#diff-05e72230161a4d41c5a27fb453041c78839c193a9ddcb5066b1cb62c5a399529R117) and pass workspace config into the function.